src/ogma/features/nodeGrouping.ts

```diff
@@ -0,0 +1,205 @@
+import {StyleRule, Transformation, Node, NodeList} from '@linkurious/ogma';
+import {ConflictValue, LkEdgeData, LkNodeData, MissingValue} from '@linkurious/rest-client';
+
+import {LKOgma} from '../index';
+import {Tools} from '../../tools/tools';
+
+export const LKE_NODE_GROUPING_EDGE = 'LKE_NODE_GROUPING_EDGE';
+
+export class NodeGroupingTransformation {
+  public transformation?: Transformation<LkNodeData, LkEdgeData>;
+  public nodeGroupingStyleRule?: StyleRule<LkNodeData, LkEdgeData>;
+  public groupRule?: {ruleName: string; type: string; property: string};
+  private _ogma: LKOgma;
+
+  constructor(ogma: LKOgma) {
+    this._ogma = ogma;
+  }
+
+  /**
+   * Set the grouping rule
+   * @param ruleName The nae of node grouping rule
+   * @param type the type of the node
+   * @param property the property name that will be used to group the nodes
+   */
+  public setGroupingRule(ruleName: string, type: string, property: string): void {
+    this.groupRule = {ruleName: ruleName, type: type, property: property};
+  }
+
+  /**
+   * create a node grouping transformation
+   * It uses groupRule to define the rule
+   * Group the nodes based on a category type and a property value
+   */
+  public async initTransformation(): Promise<void> {
+    if (this.transformation === undefined) {
```

Comment:
in which case do we trigger initTransformation with this.transformation !== undefined ?

Reply:
Normally it should never be the case. but if someone does (ogma-parser is a public library used by clients for plugins) we should just treat it as a refresh.